src/types/widget.ts

```diff
@@ -1,34 +1,18 @@
 import { EndowmentOption } from "./aws";
 
-type AdvanceOptionsDisplay = "hidden" | "collapsed" | "expanded";
-
 export type WidgetConfig = {
   endowment: EndowmentOption;
-  isDescriptionTextShown: boolean;
-  advancedOptions: {
-    display: AdvanceOptionsDisplay;
-    liquidSplitPct: number;
-  };
+  hideDescription: boolean;
+  isSplitDisabled: boolean;
+  liquidSplitPct: number;
 };
 
 export type WidgetURLSearchParams = {
-  isDescriptionTextShown: "true" | "false";
-  advancedOptionsDisplay: AdvanceOptionsDisplay;
+  hideDescription?: "true" | "false";
```

Comment:
> Do we expect users to omit some search params themselves?

I would say this is a fair (and safer) operating assumption that we should expect of users to attempt to manually modify params. Whilst _most_ users will never touch it we should anticipate that _some_ will be technically savvy enough to understand and modify/remove.

Reply:
> I would say this is a fair (and safer) operating assumption that we should expect of users to attempt to manually modify params. Whilst most users will never touch it we should anticipate that some will be technically savvy enough to understand and modify/remove.

@SovereignAndrey, this has been the assumption from the beginning - that's why we have a schema that validates the raw search param

```
1. raw search param
2. parser
3. `type WidgetSearchParam (everything is defined as this is a parsed value)`
```
What I don't understand and see as unncessary is making our parser more lax (this is the opposite of going `SAFER`). 
`FIRST`: widget url generator includes `every` attribute on generation (user has no control over this), doesn't matter if it's falsy.
```
split=50&descriptHidden=false&splitDisabled=false
```

if user tries to change this (omits some parts)
```
split=50
```
the parser would error out - instead of falling back to a default value (implicit)
```
descriptHidden required
splitDisabled required
```
but now, it would just silently default to some value (implicit) - do we make it to inform user know that incorrect missing widget attributes resolves to default values ? do we make it to inform user what the default values are - or we just let them play with url search param to see what is what. 

> be technically savvy enough to understand and modify/remove.

If we are okay with rendering the widget with no explicit attributes at all, then it would stand  to make `liquidSplit`optional too. On the other hand, given previous parser behavior, a "savvy" user would understand that removing something errors out.

Reply:
No need to get so worked up about this. Omitting boolean values is generally implied to mean that those values should be `false` - see how JavaScript's boolean checks behave with `undefined/null` or how React interprets omitted boolean props. Making these field optional just follows the same convention.

> If we are okay with rendering the widget with no explicit attributes at all, then it would stand to make `liquidSplit` optional too.

Although this is unrelated to the above optional boolean value convention, we could actually do this if you think it makes sense. It has crossed my mind, but opted not to implement that at the time.

Reply:
> Omitting boolean values is generally implied to mean that those values should be false - see how JavaScript's boolean checks behave with undefined/null or how React interprets omitted boolean props. Making these field optional just follows the same convention.

does user also know this ? Do we make it communicate to them that 
"Hey this is your generated url `a=1&b=2&c=3`, you could omit `b` & `c` if you want to, it would still work`

The inconsistency is that we design the url generator to include all this attributes but intent(all attributes are included by design) is not reflected to the parser. If you really want to change this implementation, better to reflect this also in generator. 

> see how JavaScript's boolean checks behave with undefined/null or how React interprets omitted boolean props.

that is, if attribute value is `falsy` omit it in the generated url

Reply:
@SovereignAndrey, @misicnenad, I don't really understand why we are suddenly changing the behavior of the parser (that's just doing just fine). Do we now get complaints from users like:
`"I get this widget url but when I omit something, it errored out - it says it's required"`
`" I modified the url and mispelled an attribute, it errored out - it says it's required"`

The answer is probably no, 
As a `normal` user, why would I deliberately change the widget url (remove some of it ) and still expect it to work?
While `savvy` user, if really savvy, would change something and would understand instantly why a change might work and some change errors out.

Reply:
@SovereignAndrey, @misicnenad 
To expound on to the `safety`(handling of unintended incorrect changes) aspect of this. Say a user generated a url
```
hideDescription=true&liquidSplit=60&isSplitDisabled=true
```

pasted this somewhere and for some unknown reason it gets corrupted, changed incorrectly, experimented on, to some:
```
hideDesc=true&liquidSplit=6o&isSplitDisabled=T
```

`BEFORE` errors out
```
hideDescription: required
liquidSplit: number
isSplitDisable: required
```

`NOW`, no error - just defaults to `split:50, hidden: false, splitDisabled:false` which is different to user's intent when he generated the url.

A "savvy" user would modify the original url just fine (no need to make things optional). If he makes a mistake (mispelled something, value outside of range), he would just be informed of this and would make further changes accordingly

Reply:
> that is, if attribute value is falsy omit it in the generated url

Can do that.

Reply:
> The user would go back to the widget config page, regenerate the widget snippet and hopefully not corrupt the snippet URL in the same way; 

1. how would user know that their config is corrupt/incorrect ? - doesn't matter how they messed up.
2. If they already know its corrupt/incorrect how do we tell them what to do.

`2` is not currently present (could be in another improvement PR), In addition to the validation message: `Please try to generate and copy your widget url again`


`1` is why the previous schema is strict. Going back to the example url 
```
//more practical: user want to raise liquid funds via widget
hideDescription=true&liquidSplit=100&isSplitDisabled=true
```
If they messed the config (which we already established as likely)
```
liquidsplit=100&isSplitDisable=true
```
the config generated from widget param
```
liquidSplit: 50, 
isSplitDisabled: false
hideDescription: false 
```
Instead of user getting an error (so they get informed that something went wrong), optional parser just defaulted instead which is far from the user's intent. Is this really the behavior we intend to have with this optional changes?

Reply:
We're having the same discussion in 2 threads simultaneously, see my response here https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2802#discussion_r1538872453

Reply:
@ap-justin I think the discussion on such a minor thing is getting out of hand, we had less back and forth when deciding how to implement a major important flow like the donation one.

@SovereignAndrey please be the judge in this case, I'll implement whichever approach you favor.

Reply:
Agreed that this is a VERY minor edge case. If user is curious enough to experiment with the generated URL and their widget doesn't work at all or as expected, then they should (hopefully) be astute enough to go back to the configuration page and regenerate the URL again OR they can always reach out to the support team for guidance. 

If find that we start to have users trying to change things unsuccessfully, then we could take steps to add a page to the documentation site that would break down configuring a widget outside the app, listing attributes and allowed values/ranges.

Reply:
@ap-justin can we finally put this discussion to rest now?

Reply:
Last thing I'll add is that:
- Tangentially, this optional params setup can help with backwards compatibility, should we add new options in the future, that existing widgets out in the wild could continue working unhindered, so long as we set defaults to show the widget as set originally when building in new options. 
- If we use continue to use strict params, then we'd likely need to add params/API versioning control for widgets, such that newly generated widgets in the web app would always use the latest API and config. The older API versions/configs would need to remain supported by the Lambdas.

@misicnenad Please revert to the strict setup for now, but I will open a Linear ticket to start working on aligning around how we will plan address the above idea of backwards compatibility in the widgets ASAP, which may include changing the params more comprehensively as part of that. Thank you!

Reply:
@SovereignAndrey @ap-justin I think this is all that's required to revert the change https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2802/commits/031ad27c5bb338ff13d7982687e55e0fd74bb2d6, let me know if I forgot to update anything :pray:

Reply:
@misicnenad, @SovereignAndrey 
This is a point I haven't considered when looking at optional params
> Tangentially, this optional params setup can help with backwards compatibility, 

if `splitDisabled` param is set as `required`, production widgets (if any) won't work anymore - so I guess It's prudent to have it as `optional` (if some production widget urls are in use)